src/cc/bcc_proc.c

```diff
@@ -82,6 +83,42 @@ int bcc_mapping_is_file_backed(const char *mapname) {
     STARTS_WITH(mapname, "[vsyscall]"));
 }
 
+/*
+Finds a file descriptor for a given inode if it's a memory-backed fd.
+*/
+static char *_procutils_memfd_path(const int pid, const uint64_t inum) {
+  char path_buffer[PATH_MAX + 1];
+  char *path = NULL;
+  char *dirstr;
+  DIR *dirstream;
+  struct stat sb;
+  struct dirent *dent;
+
+  snprintf(path_buffer, (PATH_MAX + 1), "/proc/%d/fd", pid);
+  dirstr = malloc(strlen(path_buffer) + 1);
+  strcpy(dirstr, path_buffer);
+  dirstream = opendir(dirstr);
+
+  if (dirstream == NULL)
+    return NULL;
```

Comment:
We didn't free `dirstr` in this error path right? Just furious why do we need copying of the string:)

Reply:
nice catch - this is definitely a memory leak. I missed it when I added the handling that @yonghong-song suggested of returning early. I'll get a quickfix up for that.

> why do we need copying of the string:)
The string is copied here so that it can be used to build the path that we return a heap-allocated reference to. Would it be better to use a local buffer to construct the directory path, and only malloc for the string we want to return? If so, I can include this change in my PR. I had considered it, but decided to just reuse the buffer - which is why we are copying here.